logic/broadcaster.js

```diff
@@ -103,6 +103,9 @@ class Broadcaster {
 	 */
 	getPeers(params, cb) {
 		params.limit = params.limit || this.config.peerLimit;
+		// Leave some room in the broadcast list for peers with a
+		// non-matching broadhash in order to avoid forks.
```

Comment:
I believe stating here as non-matching broadhash is the cause of fork is not justified. Fork occurs because broadcast does not work and sync was triggered only on the set of nodes which were on low height in the network. 

As we discussed earlier, my thoughts are that we do broadcasting to only matching boradhash peers, as its a real time and costly event across the network. For syncing we should pull blocks from any peers with the highest height in the network irrespective of the broadhash.